go-controller/pkg/clustermanager/pod/allocator.go

```diff
@@ -0,0 +1,311 @@
+package pod
+
+import (
+	"fmt"
+	"net"
+	"sync"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	listers "k8s.io/client-go/listers/core/v1"
+	"k8s.io/klog/v2"
+
+	nettypes "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/allocator/id"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/allocator/ip/subnet"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/allocator/pod"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/kube"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/types"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+)
+
+// PodAllocator acts on pods events handed off by the cluster network controller
+// and allocates or releases the pod annotation elements.
+type PodAllocator struct {
+	netInfo util.NetInfo
+
+	// ipAllocator of IPs within subnets
+	ipAllocator subnet.Allocator
+
+	// idAllocator of IDs within the network
+	idAllocator id.Allocator
+
+	// An utility to allocate the PodAnnotation to pods
+	podAnnotationAllocator *pod.PodAnnotationAllocator
+
+	// track pods that have been released but not deleted yet so that we don't
+	// release more than once
+	releasedPods      map[string]sets.Set[string]
+	releasedPodsMutex sync.Mutex
+}
+
+// NewPodAllocator builds a new PodIPAllocator
```

Comment:
this comment is outdated, isn't it ? 

FWIW, I would still call this neat struct `PodIPAllocator` though.

Reply:
Will fix

Reply:
This is the doc attached with the `PodAllocator`:
```
// PodAllocator acts on pods events handed off by the cluster network controller
// and allocates or releases resources (IPs and tunnel IDs at the time of this
// writing) to pods on behalf of cluster manager.
```
I don't think `PodIPAllocator` fits. Can take other suggestions. I thought about `PodResourceAllocator` but I didn't like it specially, as Resource can give the impression of referring to kubernetes resources, but I can bend here.

Reply:
I think that `Resource` is too generic and makes the reader think about kubernetes resources. 

Let's leave it as is then.